src/main/java/uk/gov/hmcts/reform/judicialapi/elinks/service/impl/IdamElasticSearchServiceImpl.java

```diff
@@ -220,50 +216,51 @@ public ResponseEntity<Object> getIdamDetails() {
             RefDataElinksConstants.JobStatus.IN_PROGRESS.getStatus(), IDAMSEARCH);
         Set<IdamResponse> judicialUsers = new HashSet<>();
         List<UserProfile> userProfiles = userProfileRepository.fetchObjectIdFromCurrentDate();
-        userProfiles.forEach(userProfile -> {
-            Map<String, String> params = new HashMap<>();
-            params.put("query", idamFindQuery.concat(userProfile.getObjectId()));
-            log.debug("{}:: search elk query {}", loggingComponentName, params.get("query"));
-            int count = 0;
-            int totalCount = 0;
-            ResponseEntity<Object> responseEntity = null;
-            Response response = null;
-            try {
-                String bearerToken = "Bearer ".concat(getIdamBearerToken(schedulerStartTime));
-                response = idamFeignClient.getUserFeed(bearerToken, params);
-                logIdamResponses(response);
-                responseEntity = JsonFeignResponseUtil.toResponseEntity(response,
-                    new TypeReference<Set<IdamResponse>>() {
-                    });
-                if (response.status() == 200) {
-
-                    Set<IdamResponse> users = (Set<IdamResponse>) responseEntity.getBody();
-                    judicialUsers.addAll(users);
-
-                    List<String> headerCount = responseEntity.getHeaders().get("X-Total-Count");
-                    if (headerCount != null && !headerCount.isEmpty()
-                        && !headerCount.get(0).isEmpty()) {
-
-                        totalCount = Integer.parseInt(headerCount.get(0));
-                        log.debug("{}:: Header Records count from Idam :: " + totalCount, loggingComponentName);
+        if (userProfiles.size() > 0) {
+            String bearerToken = "Bearer ".concat(getIdamBearerToken(schedulerStartTime));
+            userProfiles.forEach(userProfile -> {
```

Comment:
forEach checks for existence of elements in collection, outer 'if' statement not necessary

Reply:
Yes, foreach will do that check, but if you move this call into foreach loop, then this will call n number of times, which we are trying to avoid

Reply:
I.e. we could initialise token with:
String bearerToken = userProfiles.size() > 0? "Bearer ".concat(getIdamBearerToken(schedulerStartTime)) : "":

It would remove one level of indentation, but is otherwise same to proposed change.